packages/node/src/ethereum/ethers/json-rpc-batch-provider.spec.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable */
+import { JsonRpcBatchProvider } from './json-rpc-batch-provider';
+import { ConnectionInfo } from './web';
+
+describe('JsonRpcBatchProvider', () => {
+  let batchProvider: JsonRpcBatchProvider;
+  let fetchJsonMock: jest.SpyInstance;
+
+  beforeEach(() => {
+    // Create a new instance of the JsonRpcBatchProvider before each test
+    batchProvider = new JsonRpcBatchProvider('http://localhost:8545');
```

Comment:
Is this endpoint non-existant to test failures?

Reply:
we override the fetch method to resolve a promise locally, so no requests are made to the endpoint. it's just a placeholder